addons/stock/models/stock_picking.py

```diff
@@ -901,6 +929,15 @@ def _check_immediate(self):
                 immediate_pickings |= picking
         return immediate_pickings
 
+    def _get_empty_locations(self):
+        """ Return the locations that will be emptied out when `self` is processed. """
+        locations = self.mapped('move_line_ids.location_id')
+        empty_locations = self.env['stock.location']
+        for loc in locations:
+            if self.mapped('move_line_ids')._will_empty_location(loc):
+                empty_locations |= loc
+        return empty_locations
```

Comment:
not sure it is very efficient ? you will always pass all the move lines to _will_empty_location which will do a filtered
maybe you could group self.move_line_ids by location_id and all call _will_empty_location one time per group?